cmd/traffic/cmd/manager/mutator/watcher.go

```diff
@@ -789,16 +822,25 @@ func (c *configWatcher) Start(ctx context.Context) {
 	c.dps = make([]cache.SharedIndexInformer, len(nss))
 	c.rss = make([]cache.SharedIndexInformer, len(nss))
 	c.sss = make([]cache.SharedIndexInformer, len(nss))
+	c.rls = make([]cache.SharedIndexInformer, len(nss))
```

Comment:
I think this should be `nil` if `k8sapi.GetArgoRolloutCRDState(ctx)` returns `false`. So maybe break everything related to that condition into a separate loop.